cmd/esc/cli/env_edit.go

```diff
@@ -66,6 +62,35 @@ func newEnvEditCmd(env *envCommand) *cobra.Command {
 			}
 			_ = args
 
+			if file != "" {
+				var yaml []byte
+				switch file {
+				case "-":
+					yaml, err = io.ReadAll(env.esc.stdin)
+				default:
+					yaml, err = fs.ReadFile(env.esc.fs, file)
+				}
+				if err != nil {
+					return fmt.Errorf("reading environment definition: %w", err)
+				}
+
+				diags, err := edit.env.esc.client.UpdateEnvironment(ctx, ref.orgName, ref.envName, yaml, "")
+				if err != nil {
+					return fmt.Errorf("updating environment definition: %w", err)
+				}
+				if len(diags) == 0 {
+					fmt.Fprintln(edit.env.esc.stdout, "Environment updated.")
+					return nil
+				}
+
+				return edit.env.writeYAMLEnvironmentDiagnostics(edit.env.esc.stderr, ref.envName, yaml, diags)
```

Comment:
nit: Any particular reason to check for `len(diags) == 0` rather than `!= 0`? I was surprised that the early return was the happy path.

Reply:
This mirrors the pattern later on in the file for the editor case. The editor case takes this approach b/c the error path loops around.